src/nft/components/common/SortDropdown/SortDropdown.tsx

```diff
@@ -0,0 +1,239 @@
+import clsx from 'clsx'
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { ArrowsIcon, ChevronUpIcon, ReversedArrowsIcon } from 'nft/components/icons'
+import { buttonTextMedium } from 'nft/css/common.css'
+import { themeVars } from 'nft/css/sprinkles.css'
+import { DropDownOption } from 'nft/types'
+import { useEffect, useLayoutEffect, useMemo, useReducer, useRef, useState } from 'react'
+
+import * as styles from './SortDropdown.css'
+
+export const SortDropdown = ({
+  dropDownOptions,
+  inFilters,
+  mini,
+  miniPrompt,
+  top,
+  left,
+}: {
+  dropDownOptions: DropDownOption[]
+  inFilters?: boolean
+  mini?: boolean
+  miniPrompt?: string
+  top?: number
+  left?: number
+}) => {
+  const [isOpen, toggleOpen] = useReducer((s) => !s, false)
+  const [isReversed, toggleReversed] = useReducer((s) => !s, false)
+  const [selectedIndex, setSelectedIndex] = useState(0)
+
+  const [maxWidth, setMaxWidth] = useState(0)
+
+  const ref = useRef<HTMLDivElement>(null)
+  useOnClickOutside(ref, () => isOpen && toggleOpen())
+
+  useEffect(() => setMaxWidth(0), [dropDownOptions])
+
+  const reversable = useMemo(
+    () => dropDownOptions[selectedIndex].reverseOnClick || dropDownOptions[selectedIndex].reverseIndex,
+    [selectedIndex, dropDownOptions]
+  )
+
+  return (
+    <Box
+      ref={ref}
+      transition="250"
+      borderRadius="12"
+      borderBottomLeftRadius={isOpen ? '0' : undefined}
+      borderBottomRightRadius={isOpen ? '0' : undefined}
+      height="44"
+      style={{ width: inFilters ? 'full' : mini ? 'min' : maxWidth ? maxWidth : '300px' }}
+    >
+      <Box
+        as="button"
+        fontSize="14"
+        borderRadius="12"
+        borderStyle={isOpen && !mini ? 'solid' : 'none'}
+        background={mini ? 'none' : 'lightGray'}
+        borderColor="medGray"
+        borderWidth="1px"
+        borderBottomLeftRadius={isOpen ? '0' : undefined}
+        borderBottomRightRadius={isOpen ? '0' : undefined}
+        padding={inFilters ? '12' : mini ? '0' : '8'}
+        color="blackBlue"
+        whiteSpace="nowrap"
+        display="flex"
+        justifyContent="space-between"
+        alignItems="center"
+        width={inFilters ? 'full' : 'inherit'}
+        onClick={toggleOpen}
+        cursor="pointer"
+        className={clsx(isOpen && !mini && styles.activeDropdown)}
+      >
+        <Box display="flex" alignItems="center">
+          {isOpen || !reversable ? null : (
```

Comment:
can this be ```(isOpen || !reversable) &&```

Reply:
it should actually be `(!isOpen && reversable) &&` - ya ya boi knows  demorgans

Reply:
![image](https://user-images.githubusercontent.com/11512321/190669097-4bcb7b5e-e3be-4e25-9184-5eea11139cd3.png)